hashicorp/tfc-workflows-tooling
# chore: bump go version to 1.23

## Description

tfci's go runtime language is out of date. Additional testing and vetting is required prior to next release.

Will cut a pre-release for tfci prior to cutting release for [tfc-workflows-github](https://github.com/hashicorp/tfc-workflows-github) or [tfc-workflows-gitlab](https://github.com/hashicorp/tfc-workflows-gitlab)

## Testing

1. Checkout code locally
2. Set necessary ENV variables
3. Experiment w/ combing several commands and options
4. Verify expected behavior

Example script that can start out with after building the docker image locally:

<details open>
  <summary><b>script.sh</b></summary>

```bash
#!/bin/bash
read -p 'Workspace: ' workspacevar
read -p 'Directory: ' directoryvar
read -p 'IMG: ' imagevar

WORKSPACE=$workspacevar
Directory=$directoryvar
IMAGE=$imagevar

CV_ID=$(
  docker run --rm \
  --workdir "/tfci/workspace" \
  -v $PWD:"/tfci/workspace" \
  -e "TF_HOSTNAME" \
  -e "TF_API_TOKEN" \
  -e "TF_CLOUD_ORGANIZATION" \
  -e "TF_LOG" \
  $IMAGE tfci upload \
  --workspace=$WORKSPACE \
  --directory=$Directory \
  --json \
  | jq -r '.configuration_version_id'
)

echo $CV_ID

RUN_ID=$(
  docker run --rm \
  -e "TF_HOSTNAME" \
  -e "TF_API_TOKEN" \
  -e "TF_CLOUD_ORGANIZATION" \
  -e "TF_LOG" \
  $IMAGE tfci run create \
  --workspace=$WORKSPACE \
  --configuration_version=$CV_ID \
  --target= \
  --json \
  | jq -r '.run_id'
)

echo $RUN_ID

RES=$(
  docker run --rm \
  -e "TF_HOSTNAME" \
  -e "TF_API_TOKEN" \
  -e "TF_CLOUD_ORGANIZATION" \
  -e "TF_LOG" \
  $IMAGE tfci run apply \
  --run=$RUN_ID \
  --json \
  | jq
)

echo $RES
```

</details>
